loadimpact/k6-docs
# Feat/cloud rest api section

**Describe the changes**
This PR brings a new `Cloud REST API` section according to request. 

The following content was ported from `developers.loadimpact.com`:
- `Introduction`
- `Versioning`
- `Authentication`
- `Error handling`

The rest I skipped because I am not really sure how exactly you would like to see it, the current representation is pretty ambiguous.

**Notes**:
- Top level nav won't include this section
- There is only one link that leads to it: in `/cloud` sidebar under `Cloud REST API`. In production mode it will still lead to `developers.loadimpact.com`
- No signs of it's existing whatsoever will be available in production. It is available in `staging` env, in local development mode and in local production mod (`npm build`) given that `GATSBY_DEFAULT_DOC_URL` won't be equal to `https://k6.io/docs` - this is the mark of production. 
- I've completely excluded population of Algolia indices with content of the new section so far. I'll turn it back when we done working on `Cloud REST API`

**Steps to test**:
- pull the changes
- run the project
- follow `/cloud` section, find `Cloud REST API` in the sidebar, click
- see the `Introduction` page and completely separate sidebar with ported entries
